packages/flutter/lib/src/rendering/viewport_offset.dart

```diff
@@ -143,6 +143,10 @@ abstract class ViewportOffset extends ChangeNotifier {
   /// after [applyViewportDimension], if that method is called.
   bool applyContentDimensions(double minScrollExtent, double maxScrollExtent);
 
+  /// Updates the available semantics actions according to the the current
+  /// scroll position.
+  void updateSemanticActions();
```

Comment:
This is a really weird method to add to ViewportOffset. ViewportOffset is just an offset, it shouldn't need to know anything about accessibility.

Reply:
Maybe if you can name it something that explains when it's called, rather than what it's currently expected to do, that would work better?

Reply:
Thinking about this a little more we could just keep it an internal implementation detail of ScrollPosition (extends ViewportOffset): The ScrollPosition just registers a listener on itself and calls the private _updateSemanticActions() on itself from the listener. There shouldn't be a need to loop this outside of it's class.